docker-images/operator/scripts/dynamic_resources.sh

```diff
@@ -15,3 +15,26 @@ function get_heap_size {
     echo "${CONTAINER_HEAP_MAX}"
   fi
 }
+
+function set_xmx_xms {
+  MAX_HEAP=`get_heap_size`
+  if [ -n "$MAX_HEAP" ]; then
+      if [[ $STRIMZI_JAVA_SYSTEM_PROPERTIES == *"-Xmx"* ]]; then
+          XMX_STRING=$(echo $STRIMZI_JAVA_SYSTEM_PROPERTIES | sed -n 's/.*\(-Xmx[0-9]\+.\).*/\1/p')
+          XMX_MBYTES=$(($(echo $XMX_STRING | cut -c 5- | gawk -f ./bin/to_bytes.gawk)/1024**2))
+          if [[ $XMX_MBYTES -lt $MAX_HEAP ]]; then
+             # "Xmx set less than MAX_HEAP, using MAX_HEAP"
+             JAVA_OPTS=$(echo $JAVA_OPTS | sed s/$XMX_STRING/-Xmx${MAX_HEAP}m/)
+          fi
```

Comment:
If Xmx is set we should use that. We would only use MAX_HEAP when Xmx is _not_ set, but as discussed above, I don't think we need this function, and it certainly shouldn't be called from `launch_java.sh`.

Reply:
So... if I understand this correctly, we should just remove all this `Xms` & `Xmx` fiddling from operator images. Is that right?